### DISCUSSION TITLE
ArcGIS Javascript API 4.27 and D3.js (ver 7) - multipleDefine errors

### DISCUSSION THREAD

**Discussion by b2810s** [d3/d3#3763] [Q&A]
I have created following sample where I am loading ArcGIS Javascript API 4.27 and D3.js (ver 7) using the Script tag. 

https://codepen.io/b2810s/pen/NWebbze

This gives an error on the console related to multipleDefine. 

Any idea about this error? If I change the order of loading where D3.js is loaded first and then ArcGIS Javascript API 4.27.,The map is loaded successfully in that case. 

What is a best practice to load D3 in such cases?

**Comment by curran** [comment]
If you move the D3 tag to come first, the error goes away.

```
    <script src="https://cdn.jsdelivr.net/npm/d3@7"></script>
    <script src="https://js.arcgis.com/4.27/"></script> 
```

I suspect it's something to do with the UMD boilerplate of the D3 build conflicting with the Dojo runtime of ArcGIS.

Good luck!

**Reply by b2810s** [reply]
Thanks @curran ! Is it a bug within D3 source code? Any idea if switching order will not cause any errors with D3 code written in the application? 

Also, any alternate solution? Since both the libraries are on different screens in my application and I want to load them dynamically when it's needed.
